pkg/whiteflag/confirmation.go

```diff
@@ -28,21 +28,25 @@ type ConfirmedMilestoneStats struct {
 // then the ledger diffs are calculated, the ledger state is checked and all tx are marked as confirmed.
 func ConfirmMilestone(cachedMsBundle *tangle.CachedBundle, forEachConfirmedTx func(tx *tangle.CachedTransaction, index milestone.Index, confTime int64), onMilestoneConfirmed func(confirmation *Confirmation)) (*ConfirmedMilestoneStats, error) {
 	defer cachedMsBundle.Release()
+	msBundle := cachedMsBundle.GetBundle()
 
 	tangle.WriteLockLedger()
 	defer tangle.WriteUnlockLedger()
 
-	milestoneIndex := cachedMsBundle.GetBundle().GetMilestoneIndex()
+	milestoneIndex := msBundle.GetMilestoneIndex()
 
 	ts := time.Now()
-	confirmation, err := ComputeConfirmation(tangle.GetMilestoneMerkleHashFunc(), cachedMsBundle.Retain(), milestoneIndex)
+
+	confirmation, err := ComputeConfirmation(tangle.GetMilestoneMerkleHashFunc(), msBundle.GetTailHash())
 	if err != nil {
 		// According to the RFC we should panic if we encounter any invalid bundles during confirmation
 		return nil, fmt.Errorf("confirmMilestone: whiteflag.ComputeConfirmation failed with Error: %v", err)
 	}
+	confirmation.MilestoneIndex = milestoneIndex
```

Comment:
Don't like that we set these options in an object that was created in ComputeConfirmation

Reply:
But the COO doesn't know about its final tail hash before the bundle gets issued... So if we want to reuse the code... what would be the best practice?

Reply:
This method returns an object not containing it, and the confirmation.go creates an object with this + index + hash?